File: Application.Domain/Model/Education/EducationImport.cs
using Application.Domain.Model.Education.Dtos;

namespace Application.Domain.Model.Education
{
    public class EducationImport
    {
        public string? Id { get; set; }
        public required string FileName { get; set; }
        public string Status { get; set; } = EducationImportStatus.Pending;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? StartedAt { get; set; }
        public DateTime? CompletedAt { get; set; }
        public int Processed { get; set; }
        public int CreatedUcs { get; set; }
        public int UpdatedUcs { get; set; }
        public int Linked { get; set; }
        public string? ErrorMessage { get; set; }
        public List<CourseImportError> Errors { get; set; } = new();
    }

    public static class EducationImportStatus
    {
        public const string Pending = "Pending";
        public const string Running = "Running";
        public const string Completed = "Completed";
        public const string Failed = "Failed";
    }
}

File: Application.Client/src/app/model/education-import.ts
export interface EducationImport {
  Id: string;
  FileName: string;
  Status: string;
  CreatedAt: string;
  StartedAt?: string | null;
  CompletedAt?: string | null;
  Processed: number;
  CreatedUcs: number;
  UpdatedUcs: number;
  Linked: number;
  ErrorMessage?: string | null;
}
